src/expandableCalendar/Context/Provider.tsx

```diff
@@ -158,18 +123,15 @@ const CalendarProvider = (props: CalendarContextProviderProps) => {
 
   const renderTodayButton = () => {
     return (
-      <Animated.View style={[style.current.todayButtonContainer, {transform: [{translateY: buttonY.current}]}]}>
-        <TouchableOpacity
-          style={[style.current.todayButton, todayButtonStyle]}
-          onPress={onTodayPress}
-          disabled={isDisabled}
-        >
-          <Animated.Image style={[style.current.todayButtonImage, {opacity: opacity.current}]} source={buttonIcon}/>
-          <Animated.Text allowFontScaling={false} style={[style.current.todayButtonText, {opacity: opacity.current}]}>
-            {today.current}
-          </Animated.Text>
-        </TouchableOpacity>
-      </Animated.View>
+      <TodayButton
+        state={isToday(currentDate) ? 0 : isPastDate(currentDate) ? -1 : 1}
+        disabled={disabledButton}
+        disabledOpacity={disabledOpacity}
+        onPress={onTodayPress}
+        margin={todayBottomMargin}
+        style={todayButtonStyle}
+        theme={theme}
```

Comment:
Consider moving ALL of the logic to TodayButton. 
You can listen to the context inside the component and handle everything from there. 
It will make the Provider code way more clearer and simple

Reply:
Not sure I understand what logic you mean... Can you be more specific?

Reply:
- The `state` prop. The TodayButton can listen to currentDate and handle it internally
- Same for the onPress
- I was thinking also about the `disabled` functionality, but I see you include it in the context

Reply:
So you're saying basically to pass the TodayButton the context, right?

Reply:
you don't need to pass it, you can consume it with `useContext` inside the TodayButton code